src/server/config/__tests__/complete.js

```diff
@@ -197,4 +201,27 @@ describe('server/config completeMixin()', function () {
       sinon.assert.calledOnce(transformDeprecations);
     });
   });
+
+  describe('disabled plugins', () => {
+    it('ignores config for plugins that are disabled', () => {
+      const { callCompleteMixin } = setup({
+        settings: {
+          foo: {
+            bar: {
+              unused: true
+            }
+          }
+        },
+        disabledPluginSpecs: [
+          {
+            id: 'foo',
+            getConfigPrefix: () => 'foo.bar'
```

Comment:
__optional nit:__ `configPrefix` for the plugin `foo` will probably be `foo`, not `foo.bar`, but that's arguable :)

Reply:
yeah, I agree, but I guess an unspoken thing this test is checking is that the config prefix is used even when it's different from the plugin id